### ISSUE TITLE
Slow NHPP fitting on seqFISH+ dataset

### ISSUE THREAD

**Issue by claire20w** [jadexq/ELLA#5]
Hello,

Thank you for your work in developing ELLA! 

I’m currently running ELLA for the seqFISH+ dataset presented in the paper, using the preprocessing script provided in the repository. However, the NHPP fitting step is projected to take around 2 days. The dataset has 171 cells and 2747 genes.

Is this runtime expected for the seqFISH+ dataset, or might there be an issue with my setup? Are there any ways of speeding this process up? Please see my code running ELLA below. Thank you for your help!

```
# import ELLA
import sys
import os
import pandas as pd


ella_path = os.path.abspath(os.path.join(os.getcwd(), "..", "..", "..", "ELLA"))
sys.path.append(ella_path)


from ELLA import model_beta, model_null, loss_ll, ELLA
ella_demo = ELLA(dataset='demo')


# load data
ella_demo.load_data(data_path='input/seqfish_data_dict.pkl')
# prepare data for model fitting
ella_demo.df_registered = pd.read_pickle('output/df_registered.pkl')['df_registered']
ella_demo.nhpp_prepare()


ella_demo.nhpp_fit()
```

**Comment by jadexq** [comment]
Thank you very much for your interest! Your code looks great. If you looped over all genes one by one, though 2 days sounds a bit long but it is still within the reasonable range, so I feel your setup is likely good too. There can be 3 options to speed up the process: (1) parallel the genes, for example, run 100 genes in one job (2) the seqfish+ data has good throughput, so maybe consider using a larger learning rate and fewer iterations, for example with `adam_learning_rate_min=5e-2` and `adam_learning_rate_max=5e-2` and `max_iter=500` (see the No 12 and No 13 blocks [here](https://github.com/jadexq/ELLA/blob/main/issues/issue_3.ipynb)) (3) we are revising the code, and we aim to speed up both the registration and the nhpp fitting steps. Hope the new version will come soon in around 1-2 months. Thanks!!
